File: be_csharp_dotnet_core/Controllers/TodoController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Configuration;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Dapper;
using Npgsql;
using TodoList.Models;

namespace TodoList.Controllers
{
    [Route("/tasks")]
    [ApiController]
    public class TodoController : ControllerBase
    {

        private readonly string _connectionString;
        public TodoController(IConfiguration configuration) => _connectionString = configuration.GetConnectionString("DefaultConnection");

        // GET tasks
        [HttpGet]
        public async Task<ActionResult<Dictionary<string, dynamic>>> Get()
        {
            Dictionary<string, dynamic> tasks = new Dictionary<string, dynamic>();
            using (var connection = new NpgsqlConnection(_connectionString))
            {
                await connection.OpenAsync();
                IEnumerable<TodoTask> results = await connection.QueryAsync<TodoTask>("Select * FROM tasks ORDER BY priority asc");
                tasks.Add("tasks", results);
                tasks.Add("position", 0);
                tasks.Add("length", results.Count());
            }
            return tasks;
        }

        // POST tasks
        [HttpPost]
        public async Task<ActionResult<Dictionary<string, TodoTask>>> Post(TodoTask task)
        {
            using (var connection = new NpgsqlConnection(_connectionString))
            {
                await connection.OpenAsync();
                var sql = @"INSERT into tasks (text, priority) VALUES (@Text, @Priority)";
                await connection.ExecuteAsync(sql, task);
            }
            return new Dictionary<string, TodoTask>() {
                { "task", task }
            };
        }

        // PUT tasks
        [HttpPut()]
        public async Task<ActionResult<Dictionary<string, TodoTask>>> Put(TodoTask task)
        {
            using (var connection = new NpgsqlConnection(_connectionString))
            {
                await connection.OpenAsync();
                var sql = @"UPDATE tasks SET (text = @Text, priority = @Priority) WHERE id = @Id";
                await connection.ExecuteAsync(sql, task);
            }
            return new Dictionary<string, TodoTask>() {
                { "task", task }
            };
        }

        // DELETE tasks
        [HttpDelete()]
        public void Delete()
        {
            using (var connection = new NpgsqlConnection(_connectionString))
            {
                connection.Execute("DELETE FROM tasks;");
            }
        }
    }

    [Route("/status")]
    [ApiController]
    public class StatusController : ControllerBase
    {
        public StatusController() {}

        // GET tasks
        [HttpGet]
        public async Task<string> Get()
        {
            return "ok";
        }
    }
}

File: be_fsharp_dotnet_core/Controllers/TodoController.fs
namespace TodoList.Controllers

open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Configuration
open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Dapper
open Npgsql
open TodoList.Models

[<ApiController>]
[<Route("/tasks")>]
type TodoController (configuration:IConfiguration) =
    inherit ControllerBase()
    let _connectionString = String.Format(
                                    configuration.GetConnectionString "DefaultConnection",
                                    Environment.GetEnvironmentVariable "DEV_PG_HOST",
                                    Environment.GetEnvironmentVariable "DEV_PG_USER",
                                    Environment.GetEnvironmentVariable "DEV_PG_PASSWORD")

    [<HttpGet>]
    member __.Get() : Task<TaskList> =
        async {            
            let connection =  new NpgsqlConnection(_connectionString)
            do! Async.AwaitTask (connection.OpenAsync())
            let! results = Async.AwaitTask (connection.QueryAsync<TodoTask>("Select * FROM tasks ORDER BY priority asc"))
            let tasks = {
                Tasks=List.ofSeq results;
                Position=0;
                Length=results.Count()
            }
            do! Async.AwaitTask (connection.CloseAsync())
            return tasks
        } |> Async.StartAsTask

    [<HttpPost>]
    member __.Post(task: TodoTask) : Task<Dictionary<string,obj>> =
        async {
            let connection =  new NpgsqlConnection(_connectionString)
            do! Async.AwaitTask (connection.OpenAsync())
            let sql = @"INSERT INTO tasks (text, priority) VALUES (@Text, @Priority)"
            Async.AwaitTask (connection.ExecuteAsync(sql, task)) |> Async.RunSynchronously |> ignore
            let taskHolder = Dictionary<string, obj>()
            taskHolder.Add("task", task)
            do! Async.AwaitTask (connection.CloseAsync())
            return taskHolder
        } |> Async.StartAsTask


    [<HttpPut>]
    member __.Put(task: TodoTask) : Task<Dictionary<string,obj>> =
        async {
            let connection =  new NpgsqlConnection(_connectionString)
            do! Async.AwaitTask (connection.OpenAsync())
            let sql = @"UPDATE tasks SET text = '@Text', priority = @Priority WHERE id = @Id"
            Async.AwaitTask (connection.ExecuteAsync(sql, task)) |> Async.RunSynchronously |> ignore
            let taskHolder = Dictionary<string, obj>()
            taskHolder.Add("task", task)
            do! Async.AwaitTask (connection.CloseAsync())
            return taskHolder
        } |> Async.StartAsTask

    [<HttpDelete>]
    member __.Delete() =
        let connection =  new NpgsqlConnection(_connectionString)
        connection.Execute("DELETE FROM tasks;")

[<ApiController>]
[<Route("/status")>]
type StatusController () =
    inherit ControllerBase()

    [<HttpGet>]
    member __.Get() : Task<string> =
        async {
            return "ok"
        } |> Async.StartAsTask
